avocado/plugins/spawners/podman.py

```diff
@@ -422,12 +434,21 @@ async def update_requirement_cache(
             "commit", "-q", runtime_task.spawner_handle
         )
         image_id = stdout.decode().strip()
+        tag_prefix = self.config.get("spawner.podman.image_tag_prefix")
+        base_image = self.config.get("spawner.podman.image")
+        kind = runtime_task.task.runnable.kind
+        name = runtime_task.task.runnable.kwargs.get("name")
+        tag = f"{tag_prefix}_{base_image}_{kind}_{name}".replace(":", "_")
+        try:
+            _, _, _ = await self.podman.execute("tag", image_id, tag)
```

Comment:
What about, instead of running `podman tag` command, change the tag during the `podman commit`. IMO, If we do it by one execution instead of two, we won't need to deal with atomicity of this process when the avocado unexpectedly crashes.

I am taking about running:
 ```
podman commit -q containter_id tag
``` 
instead of:
```
podman commit -q containter_id
podman tag image_id tag
```

Reply:
Unfortunately tagging is prone to fail if there's already an image with the same tag.  And that collusion can happen for multiple reasons, including having the `requirements.sqlite` file removed without the removal of the images referred.

So, the idea here is to fail of `commit` fails, but provide the meaningful name as a convenience without increasing the possibility of functional failures.